polkadot/node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -647,19 +636,41 @@ fn has_assigned_cores(
 	false
 }
 
+fn list_of_backing_validators_in_view(
+	implicit_view: &Option<ImplicitView>,
+	per_relay_parent: &HashMap<Hash, PerRelayParent>,
+) -> Vec<AuthorityDiscoveryId> {
+	let mut backing_validators = HashSet::new();
+	let Some(implicit_view) = implicit_view else { return vec![] };
+
+	for leaf in implicit_view.leaves() {
```

Comment:
this will only return the leaves (without the allowed ancestry of these leaves)

Reply:
fixed, PTAL